plugins/package-managers/bazel/src/main/kotlin/MultiBazelModuleRegistryService.kt

```diff
@@ -61,10 +68,15 @@ internal class MultiBazelModuleRegistryService(
         /**
          * Return an exception with a message that combines the messages of all [Throwable]s in this list.
          */
-        private fun List<Throwable>.combinedException(caption: String): Throwable =
-            IllegalArgumentException(
-                "$caption:\n${joinToString("\n") { it.message.orEmpty() }}"
+        private fun Map<Int, Throwable>.combinedException(caption: String): Throwable {
+            val exceptionMessages = entries.joinToString("\n") {
+                "Registry ${it.key}: ${it.value.message.orEmpty()}"
```

Comment:
IMO that's not a very elegant approach to help debugging / improve logging. Can't we find a way to simply maintain the URL down through the place the error is logged, and straight use the URL?

Reply:
Here some sample output:

```
MultiBazelModuleRegistryService - Creating a multi Bazel module registry:
MultiBazelModuleRegistryService -   Registry 1: file://%workspace%/build/registry/
MultiBazelModuleRegistryService -   Registry 2: file://%workspace%/build/extra_registry/
MultiBazelModuleRegistryService -   Registry 3: https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main/
MultiBazelModuleRegistryService -   Registry 4: https://bcr.bazel.build
```
and
```
Bazel - Failed to fetch metadata for Bazel module 'mymodule': IllegalArgumentException: Failed to query metadata for package 'mymodule':
Registry 1: Fields [homepage, maintainers] are required for type with serial name 'org.ossreviewtoolkit.clients.bazelmoduleregistry.ModuleMetadata', but they were missing at path: $
Registry 2: Failed requirement.
Registry 3: HTTP 404 
Registry 4: HTTP 404 
```

The registries can be changed to store the URL and return is when needed. However I fear it is going to clog the logs with such messages, that would occur which each failing package:
```
Bazel - Failed to fetch metadata for Bazel module 'mymodule': IllegalArgumentException: Failed to query metadata for package 'mymodule':
Registry file://%workspace%/build/registry/: Fields [homepage, maintainers] are required for type with serial name 'org.ossreviewtoolkit.clients.bazelmoduleregistry.ModuleMetadata', but they were missing at path: $
Registry file://%workspace%/build/extra_registry/: Failed requirement.
Registry https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/: HTTP 404 
Registry https://bcr.bazel.build: HTTP 404 
```
What do you think ?

Reply:
> Here's some sample output:

I'm not questioning the look of the output, but purely the implementation. IMO it's a no-go to maintain such a map just for logging purposes.

> The registries can be changed to store the URL and return is when needed.

That sounds like the better / more elegant approach to me, as it stores things together that belong together (the registry and its URL).

> However I fear it is going to clog the logs with such messages

The output sill looks ok to me, esp. if it would be debug output, which tends to be verbose anyway.